### PR TITLE
Stable sort. No Scrubs.

### LINKED ISSUE #8: link sorting should be by chunk order and not by name
https://github.com/vijayee/js-ipfs-merkle-dag/blob/master/src/dag-node.js#L20-L22

### PR DESCRIPTION
**Pull request by vijayee**
Fixes #8 with stable sorts

### CODE REVIEW

File: src/dag-node.js [hunk 1]
```diff
@@ -52,7 +53,7 @@ function DAGNode (data, links) {
   this.addRawLink = (link) => {
     encoded = null
     this.links.push(new DAGLink(link.name, link.size, link.hash))
-    this.links.sort(linkSort)
+    stable(this.links,linkSort)
```
Comment (rht) [inline src/dag-node.js:56]: From the doc, it appears that `stable` doesn't do in-place sorting by default, unlike the built-in `Array#sort`. You have to use `stable.inplace`.

**Comment by vijayee** [comment]
My kung fu was weak. My iron fist turned out to be an iron face palm. It should be corrected now. GH-24

**Comment by daviddias** [comment]
> My iron fist turned out to be an iron face palm. 

ahahahaha

**Comment by rht** [comment]
(turned out to be this one?)

**Comment by vijayee** [comment]
Ironic face palm, lol

**Pull request merged**